assets/js/modules/analytics/hooks/useExistingTagEffect.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * Analytics useExistingTag custom hook.
+ *
+ * Sets the accountID and property if there is an existing tag.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from '../datastore/constants';
+import { STORE_NAME as CORE_MODULES } from '../../../googlesitekit/modules/datastore/constants';
+import { STORE_NAME as MODULES_TAGMANAGER } from '../../tagmanager/datastore/constants';
+const { useSelect, useDispatch } = Data;
+
+export default function useExistingTagEffect() {
+	const { setAccountID, selectProperty } = useDispatch( STORE_NAME );
+
+	const {
+		existingTag,
+		existingTagAccountID,
+		existingTagPermission,
+		gtmAnalyticsPropertyID,
+		gtmAnalyticsAccountID,
+		gtmModuleActive,
+	} = useSelect( ( select ) => {
```

Comment:
See above. Maybe worth splitting this into two different `useSelect`, one for existing tag data, one for GTM Analytics data - but I'd argue those two are independent things and don't need to be grouped.

Reply:
Updated it a little bit. Unfortunately, we can't split it into two hooks.